src/Middleware/DetectLocaleMiddleware.php

```diff
@@ -140,11 +164,11 @@ protected function getPersistLocale(HTTPRequest $request)
         }
 
         // check session then cookies
-        if ($locale = $request->getSession()->get($key)) {
+        if (static::config()->get('persist_session') && $locale = $request->getSession()->get($key)) {
```

Comment:
I think because we're assigning `$locale` a value here that we need to wrap it in its own set of brackets to ensure the logic is executed correctly